src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Unit.Tests/TypeWrapperTests.cs

```diff
@@ -198,5 +198,124 @@ public class MyType
                 receivedMembersWithAttributeNames
             );
         }
+
+        [Fact]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/58226", TestPlatforms.Browser)]
+        public void VariousGenericSerializableTypesAreSupported()
+        {
+            string source = @"
+            using System;
+            using System.Collections.Generic;
+            using System.Text.Json.Serialization;
+
+            namespace HelloWorld
+            {
+                [JsonSerializable(typeof(Dictionary<string, string>))]
+                [JsonSerializable(typeof(HelloWorld.MyClass.NestedGenericClass<string>))]
+                [JsonSerializable(typeof(HelloWorld.MyGenericClass<string>.NestedClass))]
```

Comment:
I'll also recommend test which uses double nesting to make sure we cover all scenarios

Reply:
you can just create regular test for simplicity rather than unit test, If someone regresses this behavior they will cause build break for test project